docs/configuration.asciidoc

```diff
@@ -142,6 +142,34 @@ Below is a sample `Web.config` configuration file for a ASP.NET application.
 </configuration>
 ----
 
+Additionally on ASP.NET you can also implement your own configuration reader. For this you need to implement the `IConfigurationReader` interface from the `Elastic.Apm.Config` namespace.
+
+Once you have your `IConfiguration` implementation you can use the <<config-full-framework-configuration-reader-type, FullFrameworkConfigurationReaderType>> setting.
+
+[float]
+[[config-full-framework-configuration-reader-type]]
+==== `FullFrameworkConfigurationReaderType`
```

Comment:
Once people use this, it'll kill central config - there will be no way to pull configs in this case, since they inject their own config reader. Should we point this out? 🤔

Reply:
May be i missed something in the current architecture. In my case i have an ecosystem of services and it needed to set configuration in each service. Instead of that i centralized configuration in an existing database configuration table loaded with a dedicacted 'ConfigurationReader' and the setup of APM configuration is set in a single place.

Reply:
If I'm understanding correctly, it sounds like this is used as an alternate way to single-source configuration values for all .NET agents.

We should definitely note that this will break central configuration.

Reply:
@bmorelli25 I was wrong on this one - fetching central config values happens in another component and it'll still work. "This is a feature™️"

Reply:
@kodigo35 I see - what you do sounds to me very much like a feature that we already work on. It's not fully implemented yet, meaning you cannot set all settings with that at the moment, but we constantly iterate on it and add new features. The basic idea is that you'll be able to configure your agents directly from Kibana. Look [here](https://www.elastic.co/guide/en/kibana/current/agent-configuration.html) for more info.

Reply:
Dedicated configuration reader is usefull for those who have externalized the whole configuration of their service and dont want to have extra load of work for ci/cd with new keys to set. 

Some configuration properties of agent can be done in kibana but applications should only knows apm server. this requires at least the url, service name and environment property to be set.  And if you have multiple services, applications it is not very suitable to have setting in web.config. envvar car be used too but in testing platform you can have several versions of applciations deployed and they would share the same settings.